browser.js

```diff
@@ -42,18 +43,53 @@ window.onload = function() {
     const action = isWrite ? 'write' : 'read';
     const dataToSend = isWrite ? writeData : readData;
     return async function() {
-      console.time(action);
-      for (let i=0; i<5; ++i) {
-        console.time('while');
-        console.time('send');
+      const NUM_ITER = 5;
+      const progress = log(`Running ${NUM_ITER} ${action} tests: `);
+
+      const start = window.performance.now();
+
+      // compute the mean and the standard deviation on the fly using the
+      // algorithm from TAoCP Vol 2, section 4.2.2.
+      let speedMin, speedMax, lastM, lastS;
+
+      for (let i=0; i<NUM_ITER; ++i) {
+        progress.nodeValue += '.';
+
+        const startThis = window.performance.now();
+
         const receivedData = await send(dataToSend);
-        console.timeEnd('send');
-        const message = `Finished (send ${dataToSend.length} bytes, receive ${receivedData.length} bytes)`;
-        log(message);
-        console.log(message);
-        console.timeEnd('while');
+
+        const timeThis = window.performance.now() - startThis;
+
+        // only one of those is non-zero
+        const bytesXfered = dataToSend.length + receivedData.length;
+
+        const speedThis = bytesXfered / 1000.0 / timeThis;
+
+        if (i == 0) {
+          speedMin =
+          speedMax = speedThis;
+
+          lastM = speedThis;
+          lastS = 0;
+        } else {
+          if (speedThis < speedMin)
+            speedMin = speedThis;
+          if (speedThis > speedMax)
+            speedMax = speedThis;
+
+          const oldM = lastM;
+          lastM += (speedThis - oldM)/(i + 1);
+          lastS += (speedThis - oldM)*(speedThis - lastM);
+        }
       }
-      console.timeEnd(action);
+
+      const time = window.performance.now() - start;
+
+      progress.nodeValue += ' done';
+
+      const f2 = (x) => x.toFixed(2);
+      log(`Elapsed: ${time}ms, avg speed: ${f2(lastM)}MBps, std dev ${f2(Math.sqrt(lastS/(NUM_ITER - 1)))},  min/max: ${f2(speedMin)}/${f2(speedMax)}`);
```

Comment:
```suggestion
      log(`Elapsed: ${time.toFixed(2)}ms, avg speed: ${f2(lastM)}MBps, std dev ${f2(Math.sqrt(lastS/(NUM_ITER - 1)))},  min/max: ${f2(speedMin)}/${f2(speedMax)}`);
```
Well it be better to truncate the `time` value string to shorten the message?

Reply:
I don't think we really care much about the appearances here...